truffle/src/main/ruby/core/process_mirror.rb

```diff
@@ -267,6 +269,68 @@ def convert_env_value(value)
           "a+" => ::File::RDWR   | ::File::APPEND | ::File::CREAT
         }
 
+        def spawn_setup
+          require 'fcntl'
+
+          env = options&.delete(:unsetenv_others) ? {} : ENV.to_hash
```

Comment:
I think `options` should always be a Hash. Unfortunately it seems the constructor doesn't do that, even though its documentation says it does:
```
         # Assigns @environment, @command, @argv, @redirects, @options. All
         # elements are guaranteed to be non-nil. When no env or options are
         # given, empty hashes are returned.
```
Could you fix the constructor?